src/util.h

```diff
@@ -178,77 +178,75 @@ inline TypeName* Unwrap(v8::Local<v8::Object> object);
 
 inline void SwapBytes(uint16_t* dst, const uint16_t* src, size_t buflen);
 
-class Utf8Value {
+// Allocates an array of member type T. For up to StackStorageSize items,
+// the stack is used, otherwise malloc().
+template<typename T, size_t StackStorageSize>
+class MaybeStackBuffer {
   public:
-    explicit Utf8Value(v8::Isolate* isolate, v8::Local<v8::Value> value);
-
-    ~Utf8Value() {
-      if (str_ != str_st_)
-        free(str_);
+    T* operator*() {
+      return buf_;
     }
 
-    char* operator*() {
-      return str_;
-    };
-
-    const char* operator*() const {
-      return str_;
-    };
+    const T* operator*() const {
+      return buf_;
+    }
```

Comment:
@bnoordhuis 1000 % yes, but this is basically just the `operator*` from the original classes, as it is used in ~100 places in the rest of the source code. If you’re okay with that, I’m happy to rename it (but maybe would prefer `.data()` as the name?).